modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaResult.java

```diff
@@ -17,28 +17,30 @@
 
 package org.apache.ignite.internal.replicator;
 
-import java.util.concurrent.CompletableFuture;
 import org.jetbrains.annotations.Nullable;
 
 /**
  * Represents replica execution result.
  */
 public class ReplicaResult {
+    /** Default replication outcome result. */
+    private static final CommandApplicationResult DEFAULT_RESULT = new CommandApplicationResult(null, null);
+
     /** The result. */
     private final Object res;
 
     /** The replication future. */
-    private final CompletableFuture<?> repFut;
+    private final CommandApplicationResult commandApplicationResult;
 
     /**
      * Construct a replica result.
      *
      * @param res The result.
-     * @param repFut The replication future.
+     * @param commandApplicationResult The replication result.
      */
-    public ReplicaResult(@Nullable Object res, @Nullable CompletableFuture<?> repFut) {
+    public ReplicaResult(@Nullable Object res, @Nullable CommandApplicationResult commandApplicationResult) {
         this.res = res;
-        this.repFut = repFut;
+        this.commandApplicationResult = commandApplicationResult == null ? DEFAULT_RESULT : commandApplicationResult;
```

Comment:
Should we use ReplicaResult with single param in that case?

Reply:
We can, but I prefer to keep it as is to not break anything.